device/cuda/include/traccc/cuda/seed_merging/seed_merging.hpp

```diff
@@ -0,0 +1,34 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/cuda/utils/stream.hpp"
+#include "traccc/edm/seed.hpp"
+#include "traccc/edm/nseed.hpp"
+#include "traccc/utils/algorithm.hpp"
+#include "traccc/utils/memory_resource.hpp"
+#include <vecmem/utils/copy.hpp>
+#include <vecmem/memory/unique_ptr.hpp>
+
+
+namespace traccc::cuda {
+
+class seed_merging : public algorithm<std::pair<vecmem::unique_alloc_ptr<nseed<20>[]>, uint32_t>(const seed_collection_types::buffer&)> {
```

Comment:
Why `nseed<20>`? I thought we were only going to merge 2 seeds together in this algorithm. At least for now. So why now `nseed<4>`? 😕 

At the same time the return type should really instead be `vecmem::data::vector_buffer<nseed<4> >`. Since that type is literally what you wrote here. (An array with a known size.)

Reply:
Ah okay, sorry if I did not explain properly. The five-iteration loop you commented on below means that seeds can become at most five spacepoints longer than they are now. For example, seed ABC can be merged with BCD to form ABCD in step 1, then that can merge with BCDE in step 2, etc. So In principle the seed capacity needs to be 3 + _n_ where _n_ is the number of merging steps.